libs/output-mapping/tests/Writer/TableDefinitionTest.php

```diff
@@ -258,6 +245,268 @@ public function configProvider(): Generator
+                    'name' => 'Id',
+                    'basetype' => $idDatatype->getBasetype(),
+                ],
+                [
+                    'name' => 'Name',
+                    'basetype' => $nameDatatype->getBasetype(),
+                ],
+            ],
+        ]);
+
+        $runId = $this->clientWrapper->getBasicClient()->generateRunId();
+        $this->clientWrapper->getBasicClient()->setRunId($runId);
+
+        $root = $this->tmp->getTmpFolder();
+        file_put_contents(
+            $root . '/upload/tableDefinition.csv',
+            <<< EOT
+            "1","bob","10.11","2021-12-12 16:45:21"
+            "2","alice","5.63","2020-12-12 15:45:21"
+            EOT
+        );
+        $writer = new TableWriter($this->getStagingFactory());
+
+        $tableQueue =  $writer->uploadTables(
+            'upload',
+            [
+                'typedTableEnabled' => true,
+                'mapping' => [$config],
+            ],
+            ['componentId' => 'foo'],
+            'local',
+            true
+        );
+        $jobIds = $tableQueue->waitForAll();
+        self::assertCount(1, $jobIds);
+
+        $writerJobs = array_filter(
+            $this->clientWrapper->getBasicClient()->listJobs(),
+            function (array $job) use ($runId) {
+                return $runId === $job['runId'];
+            }
+        );
+
+        self::assertCount(4, $writerJobs);
+
+        // tableColumnAdd jobs
+        $job = array_pop($writerJobs);
+        self::assertSame('tableColumnAdd', $job['operationName']);
+        self::assertSame('birthweight', $job['operationParams']['name']);
+        self::assertSame('NUMERIC', $job['operationParams']['basetype']);
+        self::assertNull($job['operationParams']['definition']);
+
+        $job = array_pop($writerJobs);
+        self::assertSame('tableColumnAdd', $job['operationName']);
+        self::assertSame('created', $job['operationParams']['name']);
+        self::assertSame('TIMESTAMP', $job['operationParams']['basetype']);
+        self::assertNull($job['operationParams']['definition']);
+
+        // modify PK job
+        $job = array_pop($writerJobs);
+        self::assertSame('tablePrimaryKeyAdd', $job['operationName']);
+        self::assertSame(['Id', 'Name'], $job['operationParams']['columns']);
+
+        // incremental import
+        $job = array_pop($writerJobs);
+        self::assertSame('tableImport', $job['operationName']);
+        self::assertSame($incrementalFlag, $job['operationParams']['params']['incremental']);
+        self::assertSame([], $job['results']['newColumns']);
+
+        $tableDetails = $this->clientWrapper->getBasicClient()->getTable($tableId);
+        self::assertTrue($tableDetails['isTyped']);
+
+        self::assertDataTypeDefinition($tableDetails['columnMetadata']['Id'], [
+            'type' => Snowflake::TYPE_NUMBER,
+            'length' => '38,0', // default integer length
+            'nullable' => true,
```

Comment:
tohle má být správně imho false, protože ten sloupec je součástí primárního klíče 🤔

Reply:
Založím bug v KBC. Je to takhle
- Založím tabulku s basetypes s PK: Metadata jsou správně non-nullable. 
- Založím tabulku s basetypes a dodatečně změnim PK: Metadata jsou nastavený nenullable.